public/app/plugins/datasource/prometheus/components/PromQueryField.tsx

```diff
@@ -166,7 +173,21 @@ class PromQueryField extends React.PureComponent<PromQueryFieldProps, PromQueryF
       range,
     } = this.props;
 
-    if (languageProvider !== prevProps.datasource.languageProvider || !_.isEqual(range, prevProps.range)) {
+    var refreshed = false;
+    if (!_.isNil(range) && !_.isNil(prevProps.range)) {
+      const absoluteRange: AbsoluteTimeRange = { from: range.from.valueOf(), to: range?.to.valueOf() };
+      const prevAbsoluteRange: AbsoluteTimeRange = {
+        from: prevProps.range.from.valueOf(),
+        to: prevProps.range.to.valueOf(),
+      };
```

Comment:
```suggestion
    let refreshed = false;
    if (range && prevProps.range) {
      const absoluteRange: AbsoluteTimeRange = { from: range.from.valueOf(), to: range.to.valueOf() };
      const prevAbsoluteRange: AbsoluteTimeRange = {
        from: prevProps.range.from.valueOf(),
        to: prevProps.range.to.valueOf(),
      };
```

What about this simplification? Is there a reason why we would have to check range and prevProps.range with `isNil`?

Reply:
Not a specific reason 🙈 Just my ignorance :) I don't know enough Typescript/Javascript to decide which one is more idiomatic.